test/metadata/main.go

```diff
@@ -0,0 +1,275 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log/slog"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/amp-labs/connectors"
+	"github.com/amp-labs/connectors/common"
+	"github.com/amp-labs/connectors/salesforce"
+	"golang.org/x/oauth2"
+)
+
+func main() {
+	creds, err := os.ReadFile("../creds.json")
+	if err != nil {
+		slog.Error("Error opening creds.json", "error", err)
+		return
+	}
+
+	var credsMap map[string]interface{}
+
+	if err := json.Unmarshal(creds, &credsMap); err != nil {
+		slog.Error("Error marshalling creds.json", "error", err)
+		return
+	}
+
+	providerApp := credsMap["providerApp"].(map[string]interface{})
+	clientId := providerApp["clientId"].(string)
+	clientSecret := providerApp["clientSecret"].(string)
+	accessToken := credsMap["AccessToken"].(map[string]interface{})["Token"].(string)
+	refreshToken := credsMap["RefreshToken"].(map[string]interface{})["Token"].(string)
+
+	salesforceSubdomain := credsMap["providerWorkspaceRef"].(string)
+
+	cfg := &oauth2.Config{
+		ClientID:     clientId,
+		ClientSecret: clientSecret,
+		Endpoint: oauth2.Endpoint{
+			AuthURL:   fmt.Sprintf("https://%s.my.salesforce.com/services/oauth2/authorize", salesforceSubdomain),
+			TokenURL:  fmt.Sprintf("https://%s.my.salesforce.com/services/oauth2/token", salesforceSubdomain),
+			AuthStyle: oauth2.AuthStyleInParams,
+		},
+	}
+
+	tok := &oauth2.Token{
+		AccessToken:  accessToken,
```

Comment:
Since expiry time is set to `time.Now().Add(-1 * time.Hour)` below, the access token will get refreshed right away, so you can actually just pass a dummy value to accessToken, there's then no need to load it from the credential file.